test/init.spec.js

```diff
@@ -0,0 +1,67 @@
+const path = require('path');
+const assert = require('yeoman-assert');
+const { skipPrettierHelpers: helpers } = require('./utils/utils');
+
+describe('JHipster init generator', () => {
+  describe('generate init', () => {
+    before(async () => {
+      await helpers.run(path.join(__dirname, '../generators/init'));
+    });
+    it('creates expected files for init generator', () => {
+      assert.file('.editorconfig');
+      assert.file('.gitattributes');
+      assert.file('.gitignore');
+      assert.file('.huskyrc');
+      assert.file('.lintstagedrc.js');
+      assert.file('package.json');
+      assert.file('.prettierignore');
+      assert.file('.prettierrc');
+      assert.file('README.md');
+      assert.file('.yo-rc.json');
```

Comment:
IMO we should migrate to snapshot.
This kind of test is not future proof.
An added file will not have a deletion protection.

Reply:
Honestly, I don't understand the behavior of snapshot. Can you explain it to me plz ?

Someone who want to add a file, should update the test first.
The test should fail, and then, the file should be added in files.js

Reply:
Add the file to files.js.
Run `npm run update-snapshots`.
or `npx mocha test/init.spec.js —updateSnapshot`